lua/line-number-change-mode/init.lua

```diff
@@ -2,28 +2,39 @@ local M = {}
 
 function M.setup(opts)
    opts = opts or {}
-
-   local group = vim.api.nvim_create_augroup("LineNumberChangeMode", {
-      clear = true
+   local va = vim.api
+   local group = va.nvim_create_augroup("LineNumberChangeMode", {
+      clear = true,
    })
 
-   vim.api.nvim_create_autocmd('ModeChanged', {
+   local function set_hl_for_mode(mode)
+      if opts.mode[mode] then
+         va.nvim_set_hl(0, "CursorLineNr", opts.mode[mode])
+
+         if mode == "c" then
+            vim.cmd.redraw()
+         end
```

Comment:
i think it's worth having a comment explaining why the redraw is necessary since it was surprising to me that it was needed

Reply:
Oops sorry.  Was there a link to documentation that spelled that out we could add?

Reply:
I didn't refer to any specific documentation, I just noticed that without the forced redraw, CursorLineNr didn't update until I typed the a character after the `:`. With the forced redraw, it updates when you hit `:`.

Reply:
@cameronr Testing it on my side and taking out the `redraw` code everything works fine.  I don't think the redraw is necessary.